docs/proposals/compute-resource-metrics-api.md

```diff
@@ -0,0 +1,141 @@
+Memory:
+    mean: 300MB (10%)
+    95th: 330MB (11%)
+    max:  395MB (13%)
+```
+
+A second user will be the scheduler. To assign pods to nodes efficiently, the
+scheduler needs to know the current free resources on each node.
+
+## Proposed endpoints
+
+	/api/v1/namespaces/myns/podMetrics/mypod
+	/api/v1/nodeMetrics/myNode
+
+The derived metrics include the mean, max and a few percentiles of the list of
+values.
+
+In the future, if any uses cases are found that would benefit from RC,
+namespace or service aggregation, metrics at those levels could also be
+exposed taking advantage of the fact that Heapster already does aggregation
+and metrics for them.
+
+Initially, this proposal included raw metrics alongside the derived metrics.
+After revising the use cases, it was clear that raw metrics could be left out
+of this proposal. They can be dealt with in a separate proposal, exposing them
+in the Kubelet API via proper versioned endpoints for Heapster to poll
+periodically.
+
+This also means that the amount of data pushed by each Kubelet to the API
+server will be much smaller.
+
+## Data gathering
+
+We will use a push based system. Each kubelet will periodically - every 10s -
+POST its derived metrics to the API server. Then, any users of the metrics can
+register as watchers to receive the new metrics when they are available.
+
+Users of the metrics may also periodically poll the API server instead of
+register as a watcher, having in mind that new data may only be available
+every 10 seconds. If any user requires metrics that are either more specific
+(e.g. last 1s) or updated more often, they should use the metrics pipeline via
+Heapster.
+
+The API server will not hold any of this data directly. For our initial
+purposes, it will hold the most recent metrics obtained from each node in
+etcd. Then, when polled for metrics, the API server would only serve said most
+recent data per node.
+
+Benchmarks will be run with etcd to see if it can keep up with the frequent
+writes of data. If it turns out that etcd doesn't scale well enough, we will
+have to switch to a different storage system.
+
+If a pod gets deleted, the API server will get rid of any metrics it may
+currently be holding for it.
+
+The clients watching the metrics data may cache it for longer periods of time.
+The clearest example would be Heapster.
+
+In the future, we might want to store the metrics differently:
+
+* via heapster - Since heapster keeps data for a period of time, we could
+  redirect requests to the API server to heapster instead of using etcd. This
+  would also allow serving metrics other than the latest ones.
+
+An edge case that this proposal doesn't take into account is kubelets being
+restarted. If any of them are, with a simple implementation they would lose
+historical data and thus take hours to gather enough information to provide
+relevant metrics again. We might want to use persistent storage directly or in
+the future to improve that situation.
```

Comment:
I think we have a redirector now: issues.k8s.io or somesuch. Since we're going to move the repo next week, let's not introduce new references to the existing location.